### PR TITLE
Delete Terraformer pod even if the context was cancelled

### PR DESCRIPTION
**Pull request by stoyanr**
**How to categorize this PR?**

/area control-plane-migration
/kind enhancements

**What this PR does / why we need it**:
Ensures that the Terraformed pod is deleted even if the context was cancelled. The context might get cancelled for example by the owner check watchdog (see #4638) and the pod should be deleted to prevent the terraform script from running after the context was cancelled.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Also removes a section with a comment that says `TODO: remove after several releases` (added in January). Besides that the code is likely not needed anymore, this gets rid of the only context-aware operation between the pod creation and waiting for it to finish, so after the pod was created there is no other exit point that might leave it running if the context was cancelled.

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: extensions/pkg/terraformer/terraformer.go [hunk 1]
```diff
@@ -255,6 +248,8 @@ func (t *terraformer) execute(ctx context.Context, command string) error {
 
 			if terminationMessage != "" {
 				podLogger.V(1).Info("Termination message of Terraformer pod: " + terminationMessage)
+			} else if ctx.Err() != nil {
+				podLogger.V(1).Info("Context error: " + ctx.Err().Error())
```
Comment (rfranzke) [inline extensions/pkg/terraformer/terraformer.go:252]: Should we still try to read the logs of the pod and fill the `terminationMessage`? It might be helpful to see what the Terraformer has done in case something goes wrong and is analyzed later on.

Comment (stoyanr) [reply extensions/pkg/terraformer/terraformer.go:252]: It can't be done with the `ctx` in this case, so a new context will have to be constructed, similarly to how it's done for `Delete`. I considered this but didn't implement it this way for the following reason: if the context is in error (cancelled or deadline exceeded), then the reason for aborting the operation is external to the terraformer and can't be found in the terraformer pod logs. The logs will be empty or "normal" and truncated at some point.

Also, I recently became aware that logs with `V(1)` are not logged by extensions by default and there is no mechanism to enable this. Even if we introduce such a mechanism in the future, to see `V(1)` logs one would need to explicitly enable them and reproduce the situation that caused the original error.

Comment (rfranzke) [reply extensions/pkg/terraformer/terraformer.go:252]: > then the reason for aborting the operation is external to the terraformer and can't be found in the terraformer pod logs. The logs will be empty or "normal" and truncated at some point.

My motivation was that the logs of the Terraform execution might be helpful in case we have to analyze issues or bugs later on - independent of the reason why the context errored. WDYT?

> Also, I recently became aware that logs with V(1) are not logged by extensions by default and there is no mechanism to enable this. Even if we introduce such a mechanism in the future, to see V(1) logs one would need to explicitly enable them and reproduce the situation that caused the original error.

True, probably the `V(1)` should be eliminated entirely then?

Comment (stoyanr) [reply extensions/pkg/terraformer/terraformer.go:252]: > My motivation was that the logs of the Terraform execution might be helpful in case we have to analyze issues or bugs later on - independent of the reason why the context errored. WDYT?

Perhaps, but at least this is not the intention in the current code. The logs are already only read if the termination message is empty. Clearly the intention is to provide a clue why the pod execution failed, not context for additional investigation.

> True, probably the V(1) should be eliminated entirely then?

Why, if this never came up as an issue so far? I think we should limit the verbosity of `Info` logs to what's really useful, and add a mechanism to increase the level to `Debug` when really needed. After discussing this with @timebertt I also included some `V(1)` logs in the watchdogs implementation with the same motivation.

In short, I still think this PR should focus on what was originally intended (making sure that the Terraformed pod is properly cleaned up if the context was cancelled), and not on changing what logs are displayed in case of failure and with what logging level.

I would also like to follow up with issues to introduce a parameter for increasing the log level for at least the provider extensions.

**Review by rfranzke — APPROVED** [review]
